homeassistant/components/history.py

```diff
@@ -304,7 +304,19 @@ def get(self, request, datetime=None):
             elapsed = time.perf_counter() - timer_start
             _LOGGER.debug(
                 'Extracted %d states in %fs', sum(map(len, result)), elapsed)
-        return self.json(result)
+
+        # Reorder the result to respect the ordering given by any
+        # entities explicitly included in the configuration.
+
+        sorted_result = []
+        for order_entity in self.filters.included_entities:
+            for state_list in result[:]:
```

Comment:
This is rather inefficient - not only does it compute a fresh copy of the list every time, it also iterates all elements every time.

Reply:
It is removing entities, so it has to iterate over a copy.

However, I can abort the loop after finding a match, since entities are only included once, as an optimization.

Reply:
Optimization committed.  No copying, minimal scanning, and no building of aux data structures.